apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -321,15 +315,37 @@ function main() {
     };
 
     $scope.send_save = function (section) {
+      const assignIdChanged = $scope.getAssignmentId(section) !== section.assign_id;
+      if (assignIdChanged) {
+        const assignable = $scope.assignable_list.find(a => a.assign_id === section.assign_id);
+        // update course/script assigned to section. Right now a section can only
+        // have one or the other, but that will change in th efuture.
+        section.script = null;
+        section.course_id = null;
+
+        if (assignable) {
+          if (assignable.is_course) {
+            section.course_id = assignable.id;
+          } else {
+            section.script = {
+              id: assignable.id,
+              name: assignable.name
+            };
+          }
+        }
+      }
```

Comment:
I suspect this isn't the most correct way of doing things in angular, where they have this idea of a model that is kept in sync between client/server. That said, the approach I've taken is one that I can understand/reason about, and will presumably also be somewhat short lived as we plan to move this table to React in the very near future.